core/modules/replica/Configuration.h

```diff
@@ -429,75 +430,610 @@ class Configuration: public ConfigurationIFace {
+     * @return A description of the newly created database family.
+     * @throw std::invalid_argument If the empty string passed as a value of the parameter,
+     *   or if the input descriptor has incorrect parameters (empty name,
+     *   0 values of the numbers of stripes or sub-stripes, or 0 value of the replication level), or
+     *   if the specified entry already exists in the configuration.
+     */
+    DatabaseFamilyInfo addDatabaseFamily(DatabaseFamilyInfo const& info);
+
+    /**
+     * Delete an existing family.
+     * @note In order to maintain consistency of the persistent state this operation will
+     *   also delete all dependent databases from both the transient and (if configured)
+     *   from the persistent store as well.
+     * @param name The name of a family.
+     * @throw std::invalid_argument If the empty string passed as a value of the parameter, or
+     *   if the specified entry was not found in the configuration.
+     */
+    void deleteDatabaseFamily(std::string const& name);
+
+    /**
+     * @param family The name of a database family.
+     * @return The minimum number of chunk replicas for a database family.
+     * @throw std::invalid_argument If the empty string passed as a value of the parameter, or
+     *   if the specified entry was not found in the configuration.
+     */
+    size_t replicationLevel(std::string const& family) const;
+
+    /**
+     * The selector for the names of the known database.
+     *
+     * @param family The optional name of a database family.
+     * @param allDatabases The optional flag which if set to 'true' will result
+     *   in returning all known database entries regardless of their PUBLISHED
+     *   status. Otherwise subset of databases as determined by the second flag
+     *   'isPublished' will get returned.
+     * @param isPublished The optional flag which is used if flag 'all' is set
+     *   to 'false' to narrow a collection of databases returned by the method.
+     * @return The names of known databases. A result of the method may be
+     *   limited to a subset of databases belonging to the specified family.
+     * @throw std::invalid_argument If the specified family (unless the empty string is passed
+     *   into the method) was not found in the configuration.
+     */
+    std::vector<std::string> databases(std::string const& family=std::string(),
+                                       bool allDatabases=false,
+                                       bool isPublished=true) const;
+
+    /**
+     * @param name The name of a database.
+     * @return 'true' if the specified database is known in the Configuration.
+     */
+    bool isKnownDatabase(std::string const& name) const;
+
+    /**
+     * @param name The name of a database.
+     * @return A database descriptor.
+     * @throw std::invalid_argument If the empty string passed as a value of the parameter, or
+     *   if the specified entry was not found in the configuration.
```

Comment:
I just would like to be precise in explaining how this (and alike) method evaluates its input data. This may be helpful for some folks, including myself, looking at the header a few months later.